src/sentry/event_manager.py

```diff
@@ -1928,6 +1938,106 @@ def _detect_performance_problems(jobs, projects):
         job["performance_problems"] = detect_performance_problems(job["data"])
 
 
+class Performance_Job(TypedDict, total=False):
+    performance_problems: Sequence[PerformanceProblem]
+
+
+@metrics.wraps("save_event.save_aggregate_performance")
+def _save_aggregate_performance(jobs: Sequence[Performance_Job], projects):
+
+    # TODO: batch operations (like rate limiting) so we don't repeat for each job
+    MAX_GROUPS = (
+        10  # safety check in case we are passed too many. constant will live somewhere else tbd
+    )
+    for idx, job in enumerate(jobs):
+        job["groups"] = []
+
+        # General system-wide option
+        rate = options.get("performance.issues.all.problem-creation")
+
+        # More granular, per-project option
+        per_project_rate = projects[idx].get_option("sentry:performance_issue_creation_rate")
```

Comment:
I'm making an assumption here that jobs and projects with the same index are related. does that make sense here? @wedamija

Reply:
Projects is a dict https://github.com/getsentry/sentry/blob/c279356cca47149a1d861bfc1abba910b6df0b07/src/sentry/event_manager.py#L361
You'd have to get the project id associated with the job instead.

Reply:
ah i saw below that we're already getting a `project` below so i moved those lines higher up
```
 event = job["event"]
 project = event.project
```